pkg/middleware/middleware.go

```diff
@@ -95,3 +98,24 @@ func AddCustomResponseHeaders(cfg *setting.Cfg) web.Handler {
 		})
 	}
 }
+
+func allowCacheControl(rw web.ResponseWriter) bool {
+	ccHeaderValues := rw.Header().Values("Cache-Control")
+
+	if len(ccHeaderValues) == 0 {
+		return false
+	}
+
+	foundPrivate := false
+	foundPublic := false
+	for _, val := range ccHeaderValues {
```

Comment:
ccHeaderValues is a string of comma separated values, so if the cache control includes a max-age this check will fail.
For example, I would expect to be able to set the following headers in the datasource resource.go:
```
 Cache-Control: private, max-age=806400
 X-Grafana-Cache: y
```
and have that pass this check.